### PR TITLE
114 pdf signer

### PR DESCRIPTION
**Pull request by matthopson**
UI for generating a signed PDF

https://github.com/flexion/ef-cms/pull/2194
Connects it to the signed document generator and uploads the files to s3.

### CODE REVIEW

**Review by sutt0n — APPROVED** [review]

File: shared/src/business/useCases/generateSignedDocumentInteractor.js [hunk 1]
```diff
@@ -9,29 +42,53 @@ const { drawImage, PDFDocumentFactory, PDFDocumentWriter } = require('pdf-lib');
...
+  } else if (sigTextData) {
+    const [timesRomanRef, timesRomanFont] = pdfDoc.embedStandardFont(
+      'Times-Roman',
+    );
+
+    page.addFontDictionary('TimesRoman', timesRomanRef);
+
+    pageContentStream = pdfDoc.createContentStream(
+      drawText(timesRomanFont.encodeText(sigTextData), {
+        font: 'TimesRoman',
```
Comment (kkoskelin) [inline shared/src/business/useCases/generateSignedDocumentInteractor.js]: I noticed line 78 shows this name hyphenated, but it's not hyphenated here.  Is that correct?

File: shared/src/business/useCases/saveSignedDocumentInteractor.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+  const caseEntity = new Case(caseRecord);
+  applicationContext.logger.timeEnd('Fetching the Case');
+
+  const originalDocumentEntity = caseEntity.documents.find(
+    document => document.documentId === originalDocumentId,
+  );
+
+  const signedDocumentEntity = new Document({
+    createdAt: new Date().toISOString(),
```
Comment (kkoskelin) [inline shared/src/business/useCases/saveSignedDocumentInteractor.js]: It's best to always use the DateHandler function 'createISODateString()' from the applicationContext

File: web-client/src/presenter/actions/completeDocumentSigningAction.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { state } from 'cerebral';
+
+export const completeDocumentSigningAction = async ({
+  applicationContext,
+  get,
+}) => {
+  const {
+    documentId: originalDocumentId,
+    pageNumber,
+    signatureData: { scale, x, y },
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/completeDocumentSigningAction.js:20]: I've never tried nested destructuring.  diabolical! :D

File: web-client/src/presenter/actions/completeDocumentSigningAction.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import { state } from 'cerebral';
+
+export const completeDocumentSigningAction = async ({
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/completeDocumentSigningAction.js:13]: lacking jsdoc

File: web-client/src/presenter/actions/setPDFPageForSigningAction.test.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { runAction } from 'cerebral/test';
+import { setPDFPageForSigningAction } from './setPDFPageForSigningAction';
+
+describe('setPDFPageForSigningAction', () => {
+  it('Sets state.pdfForSigning.pageNumber', async () => {
+    const newState = 2;
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/setPDFPageForSigningAction.test.js:6]: We might want to rename this to 'newPageNumber' for clarity

File: web-client/src/presenter/actions/setPDFSignatureDataAction.test.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import { runAction } from 'cerebral/test';
+import { setPDFSignatureDataAction } from './setPDFSignatureDataAction';
+
+describe('setPDFSignatureDataAction', () => {
+  it('Sets state.pdfForSigning.signatureData', async () => {
+    const newState = {
```
Comment (kkoskelin) [inline web-client/src/presenter/actions/setPDFSignatureDataAction.test.js:6]: `newSignatureData` perhaps?

**Review by kkoskelin — APPROVED** [review]

File: web-client/package.json [hunk 7]
```diff
@@ -95,27 +95,33 @@
...
+    "riot": "^4.2.0",
     "sinon": "^7.3.2",
     "stylelint": "^10.1.0",
     "stylelint-config-idiomatic-order": "^7.0.1",
     "stylelint-config-standard": "^18.3.0",
     "uswds": "^2.0.2",
     "wicg-inert": "^2.1.0"
+  },
+  "staticFiles": {
+    "staticPath": "node_modules/pdfjs-dist/build",
```
Comment (kkoskelin) [inline web-client/package.json]: I can see this coming in handy for a few other things.  I like it.  Is it possible to place this staticPath somewhere outside of `node_modules` though?

**Review by wbyoko — APPROVED** [review]

**Pull request merged**